3scale/tasks/main.yml

```diff
@@ -0,0 +1,149 @@
+      stringData:
+        AWS_ACCESS_KEY_ID: "$(echo $ACCESS_CREDS | cut -f 1)"
+        AWS_SECRET_ACCESS_KEY: "$(echo $ACCESS_CREDS | cut -f 2)"
+        AWS_BUCKET: "$S3_BUCKET"
+        AWS_REGION: "$REGION"
+      type: Opaque
+- name: Create APIManager in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: apps.3scale.net/v1alpha1
+      kind: APIManager
+      metadata:
+        name: apimanager-sample
+      spec:
+        wildcardDomain: 3scale.${CLUSTER_BASE_URL}
+        system:
+          fileStorage:
+            simpleStorageService:
+              configurationSecretRef:
+                name: aws-auth
+- name: Create APICast for Staging in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: apps.3scale.net/v1alpha1
+      kind: APIcast
+      metadata:
+        name: example-apicast-staging
+      spec:
+        deploymentEnvironment: staging
+        configurationLoadMode: lazy
+        adminPortalCredentialsRef:
+          name: 3scaleportal
+- name: Create APICast for Production in target_namespace
+  kubernetes.core.k8s:
+    state: present
+    namespace: "{{ target_namespace }}"
+    definition:
+      apiVersion: apps.3scale.net/v1alpha1
+      kind: APIcast
+      metadata:
+        name: example-apicast-production
+      spec:
+        adminPortalCredentialsRef:
+          name: 3scaleportal
+- name: Install and report - APIManager
+  block:
+    - name: Wait for installation of APIManager
+      kubernetes.core.k8s_info:
+        api_version: apps.3scale.net/v1alpha1
+        kind: APIManager
+        name: apimanager-sample
+        namespace: "{{ target_namespace }}"
+      register: result
+      until: result.resources[0].status.conditions.status == "True" and result.resources[0].status.conditions.type. == "Available"
+      retries: 5
+      delay: 5
+      failed_when: result.resources[0] is not defined
+  always:
+    - name: Report result for APIManager
+      ansible.builtin.set_fact:
+        assessment_result: "{{ 'pass' if (result.resources[0].status.conditions.status == 'True' and result.resources[0].status.conditions.type. == 'Available') else 'fail' }}"
+- name: Install and report - APICast for staging
+  block:
+    - name: Wait for installation of APICast for staging
+      kubernetes.core.k8s_info:
+        api_version: apps.3scale.net/v1alpha1
+        kind: APIcast
+        name: example-apicast-staging
+        namespace: "{{ target_namespace }}"
+      register: result
+      until: result.resources[0].status.conditions.ready == "True"
+      retries: 5
+      delay: 5
+      failed_when: result.resources[0] is not defined
+  always:
+    - name: Report result for APICast - Production
+      ansible.builtin.set_fact:
+        assessment_result: "{{ 'pass' if (result.resources[0].status.conditions.ready == 'True') else 'fail' }}"
```

Comment:
Since you have multiple checks for pass/fail you'll probably need to pool them together into a single AND statement.

So, register each variable, and put them all in a single block/always.